### PR TITLE
(MODULES-9191) allow puppet device command to run as --user

### PR DESCRIPTION
**Pull request by tkishel**
Make the '--user=' parameter of `puppet device` configurable.

### CODE REVIEW

File: manifests/run/via_cron/device.pp [hunk 1]
```diff
@@ -44,9 +45,15 @@
...
+      $optional_user = ''
+    } else {
+      $optional_user = " --user=${run_user}"
+    }
+
     cron { "run puppet device target ${name}":
       ensure  => $cron_ensure,
-      command => "${device_manager::run::command} ${device_manager::run::arguments} --target=${name}",
+      command => "${device_manager::run::command} ${device_manager::run::arguments} --target=${name}${optional_user}",
       user    => $::identity['user'],
```
Comment (DavidS) [inline manifests/run/via_cron/device.pp]: What is the benefit of `--user` over changing this here?

Comment (tkishel) [reply manifests/run/via_cron/device.pp]: Do you mean setting `user => $run_user` instead of `--user=${run_user}` ... or both?

Comment (DavidS) [reply manifests/run/via_cron/device.pp]: I was expecting a change that sets `user => $run_user`, so now I'm confused why you chose this route instead.

Comment (tkishel) [reply manifests/run/via_cron/device.pp]: From this comment in #46 

I'd like folks to be able to deploy puppet device configurations from a root-running agent into a unprivileged user. There are so many `user` contexts here! You just let me know what you want ...

Comment (DavidS) [reply manifests/run/via_cron/device.pp]: do you want to schedule a call next week so we can resolve this confusion?

Comment (tkishel) [reply manifests/run/via_cron/device.pp]: I'll say yes to any rationale to get together with you!

**Pull request closed without merging**